src/progress/pct/compile.p

```diff
@@ -528,11 +526,25 @@ PROCEDURE compileXref.
     END.
   END.
   ELSE DO:
-    ASSIGN c = '':U.
+    RUN logError IN hSrcProc (SUBSTITUTE("Error compiling file '&1' ...", REPLACE(ipInDir + '/':U + ipInFile, CHR(92), '/':U))).
+    EMPTY TEMP-TABLE ttErrors.
     DO i = 1 TO COMPILER:NUM-MESSAGES:
-      ASSIGN c = c + COMPILER:GET-MESSAGE(i) + '~n':U.
+      IF COMPILER:GET-NUMBER(i) EQ 198 THEN NEXT.
+      FIND ttErrors WHERE ttErrors.fileName EQ COMPILER:GET-FILE-NAME(i)
+                      AND ttErrors.rowNum   EQ COMPILER:GET-ROW(i)
+                      AND ttErrors.colNum   EQ COMPILER:GET-COLUMN(i)
+                    NO-ERROR.
+      IF NOT AVAILABLE ttErrors THEN DO:
+        CREATE ttErrors.
+        ASSIGN ttErrors.fileName = COMPILER:GET-FILE-NAME(i)
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') Statement NEXT should be on a separate line [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://sonar.riverside-software.fr/coding_rules#rule_key=rssw-oe-main%3Aeu.rssw.antlr.proparse.checks.OneStatementPerLine)